dfu: Add custom device update (#19005)
nrfconnect/sdk-nrf: doc/nrf/libraries/dfu/dfu_target.rst

   The application must schedule the upgrade of all images at once using the :c:func:`dfu_target_schedule_update` function.
   During this operation, the manifests stored in the ``dfu_partition`` partition will be processed.

> Custom upgrades
> -------------------

Reviewer: ```suggestion
Custom upgrades
---------------
```

> This type of firmware upgrade supports custom upgrades, e.g. for updating external peripherals or other custom firmware.
> The application must implement the custom upgrade logic by implementing function as defined in the :file:`include/dfu/dfu_target_custom.h` file.

Reviewer 2: ```suggestion
This firmware upgrade supports custom updates for external peripherals or other custom firmware. 
To use this feature, the application must implement the custom upgrade logic by applying the functions defined in the :file:`include/dfu/dfu_target_custom.h` file.
```
